.github/workflows/update-snapcraft.yaml

```diff
@@ -0,0 +1,44 @@
+on:
+  workflow_call:
+    inputs:
+      openstack-release:
+        description: OpenStack release
+        type: string
+        required: true
+
+jobs:
+  update-snapcraft:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: stable/${{ inputs.openstack-release }}
+      - name: Install Workflow Dependencies
+        run: pip install -r requirements-dev.txt
+      - name: Update Snapcraft Yaml
+        id: update_snapcraft
+        run: |
+          # use the script from the default branch so that the changes
+          # made to the script are not needed to be backported to stable
+          # branches
+          git fetch origin
+          git restore --source origin/${{ github.event.repository.default_branch }} tools/
+          python3 tools/update_snapcraft.py ${{ inputs.openstack-release }}
+          git diff --exit-code snap/snapcraft.yaml \
+          && echo "create_pr=0" >> $GITHUB_OUTPUT \
+          || echo "create_pr=1" >> $GITHUB_OUTPUT
+      - name: Create Pull Request
+        if: ${{ steps.update_snapcraft.outputs.create_pr == 1 }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config user.name "Github Action Bot"
+          git config user.email "<>"
+          git checkout -B automation/${{ inputs.openstack-release }}
+          git add snap/snapcraft.yaml
+          git commit -m "chore(${{ inputs.openstack-release }}): update snapcraft"
```

Comment:
nit: Can we have a better name for the pull request? E.g. "update dependencies in snapcraft"

Reply:
to keep the title, after discussion with @rgildein changed the commit message to `release(antelope): update snapcraft.yaml` in the update.